### PR TITLE
Add in message viewer

### PR DESCRIPTION
**Pull request by jcblw**
## Description

This PR adds a message viewer to the debug tools. This allows use to see information about messages that are being passed through an extensions finch-graphql. This gives great insight to time taken on certain resolvers, and also what is the payload of those resolvers.

Screen Shot 2021-05-31 at 8 31 30 PM

## Testing

- [ ] Includes test.
- [x] Manually tested.

## Additional notes

Really thinking after this feature we should probably spend some time to convert this devtools to typescript.

### CODE REVIEW

File: packages/finch-graphql-devtools/app/scripts/components/Code.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { Box } from '@chakra-ui/react'
```
Comment (mikeyamato) [inline packages/finch-graphql-devtools/app/scripts/components/Code.jsx:2]: while this isn't the extension or website, you didn't want to continue using material ui?

File: packages/finch-graphql-devtools/app/scripts/components/MessageViewer/helpers.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export const formatJSON = json => {
+  return JSON.stringify(json, null, ' ')
+}
+
+export const safeParse = json => {
```
Comment (mikeyamato) [inline packages/finch-graphql-devtools/app/scripts/components/MessageViewer/helpers.js:5]: out of curiosity, why call it `safeParse` and not something like `parseJSON`?

File: packages/finch-graphql-devtools/app/scripts/components/MessageViewer/graphql.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import gql from 'graphql-tag'
+
+export const MessagePullQueryDoc = gql`
+  query getMessages {
+    _finchDevtools {
```
Comment (mikeyamato) [inline packages/finch-graphql-devtools/app/scripts/components/MessageViewer/graphql.js:5]: underscores signify that this is an internal tool?

File: packages/finch-graphql-devtools/app/scripts/components/MessageViewer/MessageSidebar.jsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        </Box>
+      </Box>
+      <Divider />
+    </Box>
+  )
+}
+
+export const MessagesSidebar = ({ messages, selectQuery, selectedQuery }) => {
+  const renderer = renderListItem({ messages, selectQuery, selectedQuery })
+  console.log({ messages, selectQuery })
```
Comment (mikeyamato) [inline packages/finch-graphql-devtools/app/scripts/components/MessageViewer/MessageSidebar.jsx:61]: keep?

**Review by mikeyamato — APPROVED** [review]
just some comments. no blockers.

File: packages/finch-graphql-devtools/app/scripts/components/MessageViewer/MessageSidebar.jsx (hunk 4 above)
Comment (jcblw) [reply packages/finch-graphql-devtools/app/scripts/components/MessageViewer/MessageSidebar.jsx:61]: going to remove.

File: packages/finch-graphql-devtools/app/scripts/components/MessageViewer/graphql.js (hunk 3 above)
Comment (jcblw) [reply packages/finch-graphql-devtools/app/scripts/components/MessageViewer/graphql.js:5]: yep, to not take up namespace on people graph that could potentially overlap.

File: packages/finch-graphql-devtools/app/scripts/components/Code.jsx (hunk 1 above)
Comment (jcblw) [reply packages/finch-graphql-devtools/app/scripts/components/Code.jsx:2]: I'm not a huge fan of material UI, plus wanted something that was not going to add another css-in-js solution.

**Comment by jcblw** [comment]
@mikeyamato thanks for the review, going to keep this open. I think I am going to add the actual debug middleware into this PR.

File: packages/finch-graphql-devtools/app/scripts/components/MessageViewer/helpers.js (hunk 2 above)
Comment (jcblw) [reply packages/finch-graphql-devtools/app/scripts/components/MessageViewer/helpers.js:5]: safeParse indicates that when we parse it's going to not throw. JSON.parse throws with bad data so this being `safe` allows use to assume its not going to throw on bad input.

**Comment by jcblw** [comment]
Test is fixed and debugger code added.

**Pull request merged**